src/model.jl

```diff
@@ -1,3 +1,85 @@
+"""
+    TapedModel{F}
+
+State wrapper to hold `Libtask.CTask` model initiated from `f`
+"""
+struct TapedModel{F}
```

Comment:
I guess this should this also be a subtype of `AbstractModel`?
```suggestion
struct TapedModel{F} <: AbstractMCMC.AbstractModel
```